diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c03a3e01f3cfe17deb01c4d3c7018c9383a3ce3..6e59e6fa5eb3e59cd2ab84c461a34bb0cd938125 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,9 @@ find_package_from_npm(@auto_js/utility)
 add_library(auto_js OBJECT)
 target_compile_features(auto_js PUBLIC cxx_std_23)
 target_link_libraries(auto_js PUBLIC utility_js)
+if(EXPORT_IS_EXPORT)
+	target_compile_definitions(auto_js PUBLIC EXPORT_IS_EXPORT)
+endif()
 
 target_sources(auto_js
 	PUBLIC FILE_SET CXX_MODULES FILES
diff --git a/cmake/TripletUtil.cmake b/cmake/TripletUtil.cmake
index 7a91e1238633c69fc6f5250ac62e57c3b2f716ee..8e006d078b4692e6cb308a91350b06c14f93b74b 100644
--- a/cmake/TripletUtil.cmake
+++ b/cmake/TripletUtil.cmake
@@ -1,7 +1,10 @@
 # See: ./triplet_util for format
 function(triplet_util VARIABLE_NAME FORMAT)
+	set(TRIPLET_UTIL "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/triplet_util")
+	# npm strips the executable bits, so we add them back
+	file(CHMOD "${TRIPLET_UTIL}" PERMISSIONS OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 	execute_process(
-		COMMAND sh -c "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/triplet_util ${FORMAT}"
+		COMMAND sh -c "${TRIPLET_UTIL} ${FORMAT}"
 		OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE TRIPLET
 		RESULT_VARIABLE RESULT)
 	if(NOT RESULT STREQUAL 0)
